### DISCUSSION TITLE
Can I change the user/group of hdfs from clickhouse  to others?

### DISCUSSION THREAD

**Discussion by JiangXiNJU** [ByConity/ByConity#203] [Q&A] [answered]
hi，I'm trying to test cnch on hdfs. However, even if I change the owner of /root/app/usr/bin/clickhouse from clickhouse to others, the user on hdfs is still clickhouse.

Can I change the clickhouse's user on hdfs with some lightweight modifications? such as modify the xml ?

**Comment by hustnn** [comment]
@JiangXiNJU Do you have the `storage_configuration`?  You can set the hdfs path in xml file such as `server_hdfs_disk0`

```
<storage_configuration>
        <disks>
            <default></default>
            <server_local_disk1>
                <path>/xxx/server/data/</path>
            </server_local_disk1>
            <server_hdfs_disk0>
                <path>/user/clickhouse/</path>
                <type>bytehdfs</type>
            </server_hdfs_disk0>
        </disks>
        <policies>
            <default>
                <volumes>
                    <local>
                        <default>default</default>
                        <disk>default</disk>
                        <disk>server_local_disk1</disk>
                    </local>
                </volumes>
            </default>
            <cnch_default_hdfs>
                <volumes>
                    <hdfs>
                        <default>server_hdfs_disk0</default>
                        <disk>server_hdfs_disk0</disk>
                    </hdfs>
                </volumes>
            </cnch_default_hdfs>
        </policies>
    </storage_configuration>

**Comment by JiangXiNJU** [comment]
Yes, I have already changed the path to /user/zhxxx/byconity/xxx
But the user is still clickhouse.
![image](https://user-images.githubusercontent.com/22412775/229712090-7ee3ce1a-2f80-475e-9025-4701bbc9ab38.png)

**Reply by hustnn** [reply]
@JiangXiNJU Please try set `hdfs_user` in cnch_config

**Reply by Clark0** [reply, marked as answer]
> Yes, I have already changed the path to /user/zhxxx/byconity/xxx But the user is still clickhouse. ![image](https://user-images.githubusercontent.com/22412775/229712090-7ee3ce1a-2f80-475e-9025-4701bbc9ab38.png)

![image](https://user-images.githubusercontent.com/26629109/230267209-241dc04c-0645-4609-8f31-156d05470174.png) can try add hdfs_user in cnch_config.xml

**Reply by hustnn** [reply]
@JiangXiNJU is it working?

**Reply by JiangXiNJU** [reply]
Yes, it worked ! thank you very much
